### PR TITLE
Add a Security Policy

### PR DESCRIPTION
**Pull request by Wadeck**
After some discussions with @dduportal, I would like to propose this addition to set expectations about the security management of the Docker images.

Depending on the outcome, this could be re-used in other Docker repositories (like docker-inbound-agent).

Also, a link to this document could be added to https://hub.docker.com/r/jenkins/jenkins to increase visibility.

Additional context for the discovery of the information. When you are on https://www.jenkins.io/download/, if you click on "Docker", you're redirected to https://hub.docker.com/r/jenkins/jenkins, which contains a link to the README.md of this repository.

If you think about a better place for this, please don't hesitate :-)

----

- [x] Make sure you are opening from a **topic/feature/bugfix branch** (right side) and not your main branch!
- [x] Ensure that the pull request title represents the desired changelog entry
- [x] Please describe what you did
- [n/a] Link to relevant issues in GitHub or Jira
- [n/a] Link to relevant pull requests, esp. upstream and downstream changes
- [n/a] Ensure you have provided tests - that demonstrates feature works or fixes the issue

Required reviewers:
- @dduportal 

Some potentially interested reviewers:
- @daniel-beck 
- @MarkEWaite 
- @gounthar

### CODE REVIEW

File: SECURITY.md [hunk 1]
```diff
@@ -0,0 +1,31 @@
...
+- The package should have a fixed version provided in the base image that is not yet included in our image.
+
+If the finding is coming from a manual audit:
+- Please follow the process about [Reporting Security Vulnerabilities](https://jenkins.io/security/reporting/).
+
+We will reject reports coming from scanners without additional explanations.
+
+## Vulnerability Management
+
+Once the report is considered legit, a new image is published with the latest packages.
```
Comment (MarkEWaite) [inline SECURITY.md:27]:
```suggestion
Once the report is considered legitimate, a new image is published with the latest packages.
```

**Review by MarkEWaite — APPROVED** [review]
Minor optional change to use legitimate instead of the less formal "legit".  Approved whether that change is applied or not.

File: README.md [hunk 2]
```diff
@@ -348,6 +348,10 @@ Then plugins will be upgraded if the version provided by the docker image is new
 
 If you wish to contribute fixes to this repository, please refer to the [dedicated documentation](HACKING.adoc).
 
+### Security
```
Comment (lemeurherve) [inline README.md:351]:
```suggestion
## Security
```
I don't think this section should be under "Upgrading"

File: README.md [hunk 3]
```diff
@@ -348,6 +348,10 @@ Then plugins will be upgraded if the version provided by the docker image is new
...
+
+For information related to the security of the Docker image, please refer to the [dedicated documentation](README.md).
```
Comment (lemeurherve) [inline README.md:353]:
```suggestion
For information related to the security of this Docker image, please refer to the [dedicated documentation](SECURITY.md).
```

**Review by lemeurherve — COMMENTED** [review]
Small suggestions, good improvement!

File: README.md (hunk 2 above)
Comment (lemeurherve) [reply README.md:351]: I would even put it at first level (I've opened another PR to put "Hacking" and "Questions" as first level sections too as they should not be under the other ones)
```suggestion
# Security
```

File: SECURITY.md [hunk 4]
```diff
@@ -0,0 +1,31 @@
...
+
+We rely on the base image provider for the security of the system libraries.
+
+## Reporting Security Vulnerabilities
+
+Before reporting a vulnerability, here are some instructions.
+
+If the finding is coming from a Software Composition Analysis (SCA) scanner:
+- The scan must have been done on the latest version of the image.
+- The package should have a fixed version provided in the base image that is not yet included in our image.
```
Comment (daniel-beck) [inline SECURITY.md:18]: … and it needs to have existed at the time the image was created.

Or is this no longer the case?

File: SECURITY.md [hunk 5]
```diff
@@ -0,0 +1,31 @@
+# Security Policy
+
+The Jenkins project takes security seriously.
+We make every possible effort to ensure users can adequately secure their automation infrastructure.
```
Comment (daniel-beck) [inline SECURITY.md:4]: Add reference to the general security policy, explaining the below is specific to our Docker images.

File: SECURITY.md [hunk 6]
```diff
@@ -0,0 +1,31 @@
...
+
+## Docker Image Publication
+
+When an image is published, the latest image and the latest available packages are used.
+
+We rely on the base image provider for the security of the system libraries.
```
Comment (daniel-beck) [inline SECURITY.md:10]: Also mention the countless variants we have so if someone doesn't like Debian or whatever, there's plenty of others.

File: SECURITY.md [hunk 7]
```diff
@@ -0,0 +1,31 @@
...
+
+## Reporting Security Vulnerabilities
+
+Before reporting a vulnerability, here are some instructions.
```
Comment (daniel-beck) [inline SECURITY.md:14]: Weird phrasing.

**Review by timja — APPROVED** [review]

**Review by MarkEWaite — APPROVED** [review]
Thanks for the improvements.  Looks very good to me.

**Pull request merged**